core/shared/src/main/scala/zio/HasNoScope.scala

```diff
@@ -0,0 +1,12 @@
+package zio
+
+@scala.annotation.implicitNotFound(
+  """Can not prove that ${R} does not contain Scope.
+If ${R} contains a zio.Scope, please handle it explicitly. If it contains a generic type, add a context bound
+like def myMethod[R: HasNoScope](...) = ..."""
+)
+sealed trait HasNoScope[R]
+
+object HasNoScope extends HasNoScopeCompanionVersionSpecific {
+  override val instance: HasNoScope[Any] = new HasNoScope[Any] {}
```

Comment:
Can we change this to package-private?

Reply:
This will be called by the inlined code of the macro. If it is not public, the compiler will fail

Reply:
package-private methods / vals are public when compiled. Have you tried changing it to package-private and it didn't work?